ql/src/java/org/apache/hadoop/hive/ql/parse/repl/CopyUtils.java

```diff
@@ -463,7 +464,29 @@ public static Path getCopyDestination(ReplChangeManager.FileInfo fileInfo, Path
     String[] subDirs = fileInfo.getSubDir().split(Path.SEPARATOR);
     Path destination = destRoot;
     for (String subDir: subDirs) {
-      destination = new Path(destination, subDir);
+      // If the directory is created by compactor, then the directory will have the transaction id also.
+      // In case of replication, the same txn id can not be used at target, as the txn with same id might be a
+      // aborted or live txn at target.
+      // In case of bootstrap load, we copy only the committed data, so the directory with only write id
+      // can be created. The validity txn id can be removed from the directory name.
+      // TODO : Support for incremental load flow. This can be done once replication of compaction is decided.
+      if (AcidUtils.getVisibilityTxnId(subDir) > 0) {
+        if (subDir.startsWith(AcidUtils.BASE_PREFIX)) {
+          AcidUtils.ParsedBase pb = AcidUtils.ParsedBase.parseBase(new Path(subDir));
+          destination = new Path(destination, AcidUtils.baseDir(pb.getWriteId()));
```

Comment:
Looks like, we are removing the transaction id from the compactor and just leaving writeId there. Are we replicating the corresponding transaction? How do we know whether this writeId is visible. I am assuming that a failed compaction will leave a directory behind but its writeId won't be visible.

Reply:
We replicate only the visible data ..so if its a fail compaction, repl will not try to replicate that directory. So it's safe to assume that remove txn id has no impact at target cluster.

Reply:
Thanks for the explanation.